attribute/set.go

```diff
@@ -46,15 +49,22 @@ type (
 	Sortable []KeyValue
 )
 
+// Compile time check these types remain comparable.
+var (
+	_ = isComparable(Set{})
```

Comment:
"tracers", "aggregations", "loggers" currently are stored in`map[instrumentation.Scope]XYZ`. However, `instrumentation.Scope` has `attribute.Set` field. This means this PR would be these maps would be "safe to use" as currently they should use `attribute.Distinct`. This is for me another argument for this change.